deploy/transport-controller-deployment.yaml.template

```diff
@@ -0,0 +1,89 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: transport-controller-config
+  namespace: ${WDS_NAME}-system
+data:
+  get-kubeconfig.sh: |
+    #!/bin/bash
+    # this script receives a ControlPlane name and a parameter
+    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
+    # or the external kubeconfig (if set to "true", the first will be retrieved).
+    # The function returns the requested kubeconfig's content in base64.
+    # it assumes that it runs within a kubeflex context.
+
+    cpname="$1"
+    get_incluster_key="$2"
+
+    key=""
+    if [[ $get_incluster_key == "true" ]]; then
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
+    else
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+    fi
+
+    # get secret details
+    secret_name=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.name}')
+    secret_namespace=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.namespace}')
+
+    # get the kubeconfig in base64
+    kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: transport-controller
+  namespace: ${WDS_NAME}-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: transport-controller
+  template:
+    metadata:
+      labels:
+        name: transport-controller
+    spec:
+      serviceAccountName: kubestellar-controller-manager
+      initContainers:
+      - name: setup-wds-kubeconfig
+        image: quay.io/kubestellar/kubectl
+        imagePullPolicy: Always
+        command: [ "bin/sh", "-c", "bash /mnt/config/get-kubeconfig.sh ${WDS_NAME} true | base64 -d > /mnt/shared/wds-kubeconfig"]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /mnt/config
+        - name: shared-volume
+          mountPath: /mnt/shared
+      - name: setup-imbs-kubeconfig
+        image: quay.io/kubestellar/kubectl
+        imagePullPolicy: Always
+        command: [ "bin/sh", "-c", "bash /mnt/config/get-kubeconfig.sh ${IMBS_NAME} false | base64 -d > /mnt/shared/transport-kubeconfig"]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /mnt/config
+        - name: shared-volume
+          mountPath: /mnt/shared
+      containers:
+        - name: transport-controller
+          image: ${TRANSPORT_CONTROLLER_IMAGE}
+          imagePullPolicy: ${IMAGE_PULL_POLICY}
+          args:
+          - --transport-kubeconfig=/mnt/shared/transport-kubeconfig
+          - --wds-kubeconfig=/mnt/shared/wds-kubeconfig
+          - --wds-name=${WDS_NAME}
+          - -v=4
+          volumeMounts:
+          - name: shared-volume
+            mountPath: /mnt/shared
+            readOnly: true
+      hostAliases:
+      - ip: ${HOST_IP}
+        hostnames: [ "wds1.localtest.me", "imbs1.localtest.me" ]
+      volumes:
```

Comment:
it looks like these values are hardcoded ... they should be parametrized, no ? we could use other names for the wds/imbs and there could be multiple wds...

Reply:
additionally, this may work for local testing but what about deployment in OCP, is `hostAliases` workable there?

Reply:
I am also wondering ... why those hostnames are needed at all ? The pod should always be using the in-cluster config, unless you intend to run the pod on a cluster different from where wds and imbs are located.

Reply:
I thought we already agreed we are using the "Inventory Transport Space (ITS)" terminology rather than "IMBS". I know some stuff has not been updated, but that does not undo the decision. I would not harden the mistake by hard-coding it in something in git.

Reply:
all e2e tests are currently using the name `imbs`, I was just using the same pattern.
I agree with the comment, but this can be done in a separate PR (after the release) to align name to ITS in all places. 
btw, after introducing inventory, it will be renamed (again) to transport space.
so we may end up with:
wds, transport space, inventory space.
and transport+inventory can run on the same space, so it may become ITS.

Reply:
@francostellari @pdettori before adding the alias I couldn't run the controller using a local image loaded kubeflex.
this is a crucial part in being able to run e2e for code changes of KS repo.
keep in mind that alias is NOT replacing the name. it's just an alias (another name), which allows us to run in case of local env. If I'm not mistaken, I don't think there should be any problem with running in any cluster.

but you are correct that I missed the parameterization of this (probably when I tested locally). will push a fix.

Reply:
@pdettori regarding the fact that there are multiple wds - you are correct but it's not relevant to this deployment, since transport is per wds. this specific transport controller deployment is per specific WDS. so we should use the WDS_NAME variable.

Reply:
fixed to use parameters instead of hardcoded.

Reply:
I don't think we use `imbs`  as a name. I agree that the current controller has assumption of a single imbs, but I think it searches KF for a space with imbs label so there is flexibility in the name